### PR TITLE
qa: use centos9 for cephfs stock kernel testing

### PR DESCRIPTION
**Pull request by batrick**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by cbodley** [comment]
jenkins test api

File: qa/distros/all/rhel_9.yaml [hunk 1]
```diff
@@ -0,0 +1 @@
+rhel_9.3.yaml
```
Comment (vshankar) [inline qa/distros/all/rhel_9.yaml:1]: @kotreshhr ran into an issue - https://pulpito.ceph.com/khiremat-2024-01-19_06:49:40-fs:workload-wip-khiremat-49945-referent-inode-2-distro-default-smithi/7521650/

```
Error reimaging machines: Fog has no smithi_rhel_9.3 image. Available smithi images: ['smithi_centos_8.stream', 'smithi_centos_9.stream', 'smithi_rhel_8.2', 'smithi_rhel_8.4', 'smithi_rhel_8.6', 'smithi_ubuntu_16.04', 'smithi_ubuntu_18.04', 'smithi_ubuntu_20.04', 'smithi_ubuntu_22.04']
```

Temp fog issue or missing image?

Comment (batrick) [reply qa/distros/all/rhel_9.yaml:1]: oops, I guess rhel9 is not in fog yet.

Comment (batrick) [reply qa/distros/all/rhel_9.yaml:1]: We'll use centos9 instead!

Comment (batrick) [reply qa/distros/all/rhel_9.yaml:1]: https://tracker.ceph.com/issues/64148

Comment (vshankar) [reply qa/distros/all/rhel_9.yaml:1]: Yeh.

@cbodley -- I can test the fs suite with this change along with #53901 once the changes are ready.

File: qa/distros/all/rhel_9.0.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+os_type: rhel
```
Comment (vshankar) [inline qa/distros/all/rhel_9.0.yaml:1]: Is there a need to still have the rhel yamls with this change?

Comment (batrick) [reply qa/distros/all/rhel_9.0.yaml:1]: No, but I felt it was harmless to include them for the future. I can remove the commit if you like.

Comment (vshankar) [reply qa/distros/all/rhel_9.0.yaml:1]: Let's add these yamls where we have rhel9 fog images.

Comment (batrick) [reply qa/distros/all/rhel_9.0.yaml]: done

**Review by rishabh-d-dave — APPROVED** [review]
LGTM.

**Comment by vshankar** [comment]
jenkins test api

**Comment by dparmar18** [comment]
are we going to change this to use rhel9 once we have it in fog?

**Review by vshankar — APPROVED** [review]

File: qa/cephfs/mount/kclient/overrides/distro/stock/centos_9.stream.yaml [hunk 3]
```diff
@@ -0,0 +1 @@
+.qa/distros/all/centos_9.stream.yaml
```
Comment (idryomov) [inline qa/cephfs/mount/kclient/overrides/distro/stock/centos_9.stream.yaml:1]: Just wondering, isn't centos 9 already covered by another alternative in this directory, `qa/cephfs/mount/kclient/overrides/distro/stock/k-stock.yaml`?  Or is the intent to do more runs with centos 9 than with ubuntu 22.04?

Comment (batrick) [reply qa/cephfs/mount/kclient/overrides/distro/stock/centos_9.stream.yaml:1]: The intent of this yaml is to only do stock kernel testing with centos9. The main reason for this was that we can actually downstream fixes for that kernel in a predictable manner.

Comment (idryomov) [reply qa/cephfs/mount/kclient/overrides/distro/stock/centos_9.stream.yaml:1]: I see, I missed that `k-stock.yaml` is not an alternative but rather the snippets get combined there.  Thanks!

**Pull request merged**